cmd/skopeo/list_tags.go

```diff
@@ -113,23 +114,48 @@ func (opts *tagsOptions) run(args []string, stdout io.Writer) (retErr error) {
 		return fmt.Errorf("Invalid %q: does not specify a transport", args[0])
 	}
 
-	if transport.Name() != docker.Transport.Name() {
-		return fmt.Errorf("Unsupported transport '%v' for tag listing. Only '%v' currently supported", transport.Name(), docker.Transport.Name())
-	}
-
-	// Do transport-specific parsing and validation to get an image reference
-	imgRef, err := parseDockerRepositoryReference(args[0])
-	if err != nil {
-		return err
-	}
-
 	var repositoryName string
 	var tagListing []string
-	if err = retry.RetryIfNecessary(ctx, func() error {
-		repositoryName, tagListing, err = listDockerTags(ctx, sys, imgRef)
-		return err
-	}, opts.retryOpts); err != nil {
-		return err
+
+	if transport.Name() == docker.Transport.Name() {
+		// Do transport-specific parsing and validation to get an image reference
+		imgRef, err := parseDockerRepositoryReference(args[0])
+		if err != nil {
+			return err
+		}
+		if err = retry.RetryIfNecessary(ctx, func() error {
+			repositoryName, tagListing, err = listDockerTags(ctx, sys, imgRef)
+			return err
+		}, opts.retryOpts); err != nil {
+			return err
+		}
+	} else if transport.Name() == archive.Transport.Name() {
+		parts := strings.SplitN(args[0], ":", 2)
```

Comment:
While parsing works like that just fine, I'd prefer using the transport's API directly.

```go
ref, err := dockerArchiveTransport.ParseReference(args[0])         
if err != nil {                                                 
        return err
}                            
archivePath := ref.StringWithinTransport()
```